test/validation/api/ipsec/ipsec_test_in.c

```diff
@@ -7,8 +7,49 @@
 
 #include "ipsec.h"
 
+#include "reass_test_vectors.h"
 #include "test_vectors.h"
 
+static void part_prep_esp(ipsec_test_part part[], int num_part, bool v6_tunnel)
+{
+	int i;
+
+	memset(part, 0, sizeof(ipsec_test_part) * num_part);
+
+	for (i = 0; i < num_part; i++) {
+		part[i].num_pkt = 1;
+
+		if (v6_tunnel)
+			part[i].out[0].l3_type = ODP_PROTO_L3_TYPE_IPV6;
+		else
+			part[i].out[0].l3_type = ODP_PROTO_L3_TYPE_IPV4;
+
+		part[i].out[0].l4_type = ODP_PROTO_L4_TYPE_ESP;
+	}
+}
+
+static void part_prep_plain(ipsec_test_part part[], int num_part, int num_pkt, bool v6, bool udp)
+{
+	int i, j;
+
+	memset(part, 0, sizeof(ipsec_test_part) * num_part);
+
+	for (i = 0; i < num_part; i++) {
+		part[i].num_pkt = num_pkt;
+		for (j = 0; j < num_pkt; j++) {
```

Comment:
This function is never called with num_part or num_pkt other than 1 so maybe the arguments and the looping based on them could just be dropped. Such code can be added when actually needed.

Reply:
I was making it similar to part_prep_esp(). But then, agreed on the comment. Will address in next version.